kiwi/system/kernel.py

```diff
@@ -109,14 +114,16 @@ def copy_kernel(self, target_dir, file_name=None):
         """
         kernel = self.get_kernel()
         if kernel:
-            if not file_name:
+            if file_name is None:
```

Comment:
This is a behavior change. ```not str``` matches empty strings and None. ```str is None``` only matches if the string is of the None type. An empty string ```str=""``` would not be matched. Any particular reason for this change ?

Reply:
I missed the empty string check but the reason for the change is that when deciding the changes for strong typing I got an
error asking for explicit check in those lines. I will revert it as it is wrong and not explicit check needed anymore